codepropertygraph/src/main/scala/io/shiftleft/passes/ParallelCpgPass.scala

```diff
@@ -271,17 +290,33 @@ abstract class ConcurrentWriterCpgPass[T <: AnyRef](cpg: Cpg, outName: String =
               baseLogger.debug("Shutting down WriterThread")
               terminate = true
             case Some(diffGraph) =>
-              val appliedDiffGraph = DiffGraph.Applier.applyDiff(diffGraph, cpg, withInverse, keyPool)
-              if (doSerialize) {
-                store(serialize(appliedDiffGraph, withInverse),
-                      generateOutFileName(prefix, outName, index),
+              val listener =
+                if (withInverse) new BatchUpdateInverseListener
+                else if (doSerialize) new BatchUpdateForwardListener
+                else null
+
+              nDiffT += overflowdb.BatchedUpdate
+                .applyDiff(cpg.graph, diffGraph, keyPool.getOrElse(null), listener)
```

Comment:
If this was scala, I'd agree. The new BatchedUpdate API in odb is java, though, where `null` is the normal way of encoding optional non-primitive values.

Reply:
java has `java.util.Optional` since java8, and we use it in many places already. it just helps to clarify your intent: this thing is optional. 
if you really don't like it, then how about the secondary constructor idea?